### ISSUE TITLE
How to assign a fraction of GPU in a multi-GPU setting for a ray actor ?

### ISSUE THREAD

**Issue by shamanez** [ray-project/ray#14101]
I am using Ray actor to asyncronously calculate embedding from pre-trained BERT models and keep updating a knowledgebase inside a multi-GPU training process.  

Right now I have to detach the BERT model from GPU in order to execute in a RAY worker. It says there are not  "CUDA" decides. But having able to use GPU can drastically improve the execution time.  The following code shows the initialization of my RAY actor. Please let me know how to add a GPU to it.
```

            remote_cls_saver = ray.remote(RayEncode)  #saver class
            saver_actor = [  #single actor for calculate embeddings and save them
                remote_cls_saver.options(name="embedding_saver_{}".format(i)).remote() 
                for i in range(1)
            ]  
```

**Comment by richardliaw** [comment]
Maybe try `remote_cls_saver.options(name..., num_gpus=0.1)`

**Comment by shamanez** [comment]
ok, I will update.

**Comment by shamanez** [comment]
@richardliaw 

```

 # Create Ray actors only for rank 0.
        if ("LOCAL_RANK" not in os.environ or os.environ["LOCAL_RANK"] == 0) and (
            "NODE_RANK" not in os.environ or os.environ["NODE_RANK"] == 0
        ):
            
            remote_cls = ray.remote(RayRetriever)
            named_actors = [
                remote_cls.options(name="retrieval_worker_{}".format(i)).remote() 
                for i in range(args.num_retrieval_workers)
            ]


            

 
                     
        else:
            logger.info(
                "Getting named actors for NODE_RANK {}, LOCAL_RANK {}".format(
                    os.environ["NODE_RANK"], os.environ["LOCAL_RANK"]
                )
            )
            named_actors = [ray.get_actor("retrieval_worker_{}".format(i)) for i in range(args.num_retrieval_workers)]
     

```
This is how I initialized the ray cluster according to this huggingface [code](https://github.com/huggingface/transformers/blob/master/examples/research_projects/rag/finetune_rag.py#L544).

As you can see it get created with GPU device 0 and, kind of gets copied in other devices. Now I want to execute this worker in a preferred GPU. Let's say GPU ranked 4th. How to do this?

**Comment by shamanez** [comment]
any update?
